aws/src/main/java/org/apache/iceberg/aws/s3/S3OutputStream.java

```diff
@@ -175,8 +175,15 @@ private void newStream() throws IOException {
       stream.close();
     }
 
+    boolean createStagingDirectory = false;
+    if (!stagingDirectory.exists()) {
+      createStagingDirectory = stagingDirectory.mkdirs();
+    }
```

Comment:
nit: newline needed after a control statement

Reply:
Addressed. Also, separated out the logic in a method: `createStagingDirectoryIfNotExists()`